tests/queries/0_stateless/02302_s3_file_pruning.sql

```diff
@@ -0,0 +1,35 @@
+-- Tags: no-fasttest
+-- Tag no-fasttest: Depends on AWS
+
+-- { echo }
+drop table if exists test_02302;
+create table test_02302 (a UInt64) engine = S3(s3_conn, filename='test_02302_{_partition_id}', format=Parquet) partition by a;
+insert into test_02302 select number from numbers(10) settings s3_truncate_on_insert=1;
+select * from test_02302;  -- { serverError 48 }
+drop table test_02302;
+
+set max_rows_to_read = 1;
+
+-- Test s3 table function with glob
+select * from s3(s3_conn, filename='test_02302_*', format=Parquet) where _file like '%5';
+
+-- Test s3 table with explicit keys (no glob)
+-- TODO support truncate table function
```

Comment:
In order to construct an S3 storage with explicit keys, I have to apply the following clusmy steps. Things can be improved:

1. allow to do `truncate table function s3(...)`
2. allow to create an S3 storage with a list of keys (or even different buckets)